core/src/main/scala-3/org/http4s/syntax/LiteralsSyntax.scala

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s
+package syntax
+
+import scala.quoted._
+
+trait LiteralsSyntax {
```

Comment:
Is this a `trait` so we can extend it in the package object?

Reply:
yes, we need this trait, as the scala source dir references it. It can however be empty. 
It just needs to exist. Since we can have top level extensions, we dont really need this anymore I guess.

Reply:
It might be nice to make the various macro functions, e.g. `validateUri` etc, `private` so we don't expose them in the public API. I noticed that Scala3 doesn't let you call private methods on a companion object in a macro implementation (I'm unclear why), but if we lift them out of the trait, effectively making them private to the package object `literals` we should be able to hide them from the public API.

This could pose problems since we actually export these twice, e.g.`all`/`literals`, but I _think_ we can use the new `export` feature to `export` them into the `all` object, though I've not tested that.

Reply:
In log4s, we used top-level definitions in `scala-3`.  It's the only place we _can_ fully embrace Scala 3.